packages/backfill/src/index.ts

```diff
@@ -54,73 +54,87 @@ export async function backfill(
 }
 
 export async function main(): Promise<void> {
-  const config = createConfig();
-  const {
-    cacheStorageConfig,
-    internalCacheFolder,
-    logFolder,
-    outputFolder,
-    packageRoot,
-    performanceReportName,
-    clearOutputFolder,
-    logLevel,
-    hashGlobs
-  } = config;
+  try {
+    const config = createConfig();
+    const {
+      cacheStorageConfig,
+      clearOutputFolder,
+      hashGlobs,
+      internalCacheFolder,
+      logFolder,
+      logLevel,
+      mode,
+      outputFolder,
+      packageRoot,
+      performanceReportName
+    } = config;
 
-  if (logLevel) {
-    setLogLevel(logLevel);
-  }
+    if (logLevel) {
+      setLogLevel(logLevel);
+    }
 
-  const helpString = "Backfills unchanged packages.";
-
-  const argv = yargs
-    .strict()
-    .usage(helpString)
-    .alias("h", "help")
-    .version(false)
-    .option("generate-performance-report", {
-      description: "Generate performance report",
-      type: "boolean"
-    })
-    .option("audit", {
-      description: "Compare files changed with those cached",
-      type: "boolean"
-    }).argv;
-
-  const buildCommand = createBuildCommand(
-    argv["_"],
-    clearOutputFolder,
-    outputFolder
-  );
-
-  const cacheStorage = getCacheStorageProvider(
-    cacheStorageConfig,
-    internalCacheFolder
-  );
+    const helpString = "Backfills unchanged packages.";
+
+    const argv = yargs
+      .strict()
+      .usage(helpString)
+      .alias("h", "help")
+      .version(false)
+      .option("generate-performance-report", {
+        description: "Generate performance report",
+        type: "boolean"
+      })
+      .option("audit", {
+        description: "Compare files changed with those cached",
+        type: "boolean"
+      }).argv;
+
+    if (mode !== "READ_WRITE") {
+      logger.info(`Running in ${mode} mode.`);
+    } else {
+      logger.verbose(`Running in ${mode} mode.`);
+    }
 
-  const hasher = new Hasher({ packageRoot }, getRawBuildCommand());
+    const buildCommand = createBuildCommand(
+      argv["_"],
+      clearOutputFolder,
+      outputFolder
+    );
 
-  if (argv["generate-performance-report"]) {
-    await logger.generatePerformanceReport(logFolder, performanceReportName);
+    if (mode === "PASS") {
+      try {
+        await buildCommand();
+      } catch (err) {
+        throw new Error("Command failed");
```

Comment:
should we print content of 'err' to the user?

Reply:
So, the buildCommand will output stderr. Wouldn't printing the error here as well repeat it, or would it provide additional value?

Reply:
I guess it depends if the command log things to stderr before throwing, it may not. Maybe we can assume it will???

Reply:
It will. I actually have a test for it :)